### PR TITLE
Customizeable max breath for players

### PR DESCRIPTION
**Pull request by SmallJoker**
Implements requested feature of #1972.

### CODE REVIEW

**Comment by C1ffisme** [comment]
Shouldn't it be "maximum"?

File: src/object_properties.cpp [hunk 1]
```diff
@@ -153,4 +155,6 @@ void ObjectProperties::deSerialize(std::istream &is)
 	infotext = deSerializeString(is);
 	wield_item = deSerializeString(is);
 	can_zoom = readU8(is);
+	if (is.good())
+		breath_max = readU16(is);
```
[the remark this answers is not in the document]
Comment (nerzhul) [reply src/object_properties.cpp]: both ways are good, try catch is easier to use else we will need if (is.good()) on each new read

**Comment by Wuzzy2** [comment]
While you're at it, could you pleae also make max. health settable? (this would fix https://github.com/minetest/minetest/issues/2246)
Or would that be way too different?

**Comment by SmallJoker** [comment]
@Wuzzy2 HTTP 303: #6287

Rebasing..

File: src/object_properties.cpp [hunk 2]
```diff
@@ -155,5 +157,9 @@ void ObjectProperties::deSerialize(std::istream &is)
 	infotext = deSerializeString(is);
 	wield_item = deSerializeString(is);
 	can_zoom = readU8(is);
-	glow = readS8(is);
+
+	if (is.good()) {
+		glow = readS8(is);
```
Comment (nerzhul) [inline src/object_properties.cpp]: why adding glow to the is.good ?

Comment (SmallJoker) [reply src/object_properties.cpp]: It was added yesterday, where this check was forgotten. I'm moving both into a try/catch to ensure there are no errors. Indeed, `is.good` is really not suited for these compatibility checks.

**Comment by nerzhul** [comment]
@rubenwardy exceptions are slow when triggered, the catching is slow, if not triggered there is no overhead

**Pull request merged**